### DISCUSSION TITLE
ism7mqtt terminates or aborts when ism7 connectivity problems occur

### DISCUSSION THREAD

**Discussion by PITZH** [zivillian/ism7mqtt#140] [Ideas]
I managed to install ism7mqtt and via node-red and influxdb I present the data in grafana dashboards. Very cool.

The connectivity of the ism7 module is not very stable - various problems reported in forums, but not really resolved. Sometimes it happens every few minutes, sometimes it works for hours. The connectivity also impacts ism7mqtt. The program terminates if while it is running the ism7 connection falls down. This means you have to restart ism7mqtt - on my Raspberry I run a script with a simple loop:
while true
do
  ism7mqtt ... parameters ...
  sleep 30
enddo

This works in principle however when ism7 connectivity is down, ism7mqtt starts again it aborts if connectivity is still missing. The aborts lets my script ending as well.

From my view it would be great if there would be some kind of lost-connectivity-tolerance in ism7mqtt - if connection is lost it just tries again, but would not end or even abort. The behaviour could also by managed via a command line parameter.

Anyone experiencing the same problem? How did you resolve it?

**Comment by zivillian** [comment]
Have you tried running it via [systemd](https://github.com/zivillian/ism7mqtt/blob/29f9e43de8ef65711cad0c2a32905730e4cbf991/ism7mqtt.service)?

**Comment by PITZH** [comment]
was not on my radar (due to lack of knowledge...). I configured it using the service file you provided and it works right now - will check after 24 hours
